staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/streaming_test.go

```diff
@@ -0,0 +1,318 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package json
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	fuzz "github.com/google/gofuzz"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	testapigroupv1 "k8s.io/apimachinery/pkg/apis/testapigroup/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestStreamingCollectionsEncoding(t *testing.T) {
+	var streamingBuffer bytes.Buffer
+	var normalBuffer bytes.Buffer
+	normalSerializer := NewSerializerWithOptions(DefaultMetaFactory, nil, nil, SerializerOptions{StreamingCollectionsEncoding: false})
+	var remainingItems int64 = 1
+	for _, tc := range []struct {
```

Comment:
Seems we didn't cover the fallback case for non-list object
Maybe could add one for it to ensure fallback works. 

```
        {
            name:           "Non-list object, fallback expected",
            in:             &testapigroupv1.Carp{ObjectMeta: metav1.ObjectMeta{Name: "not-a-list"}},
            expectFallback: true,
        },
...
            if tc.expectFallback {
                // We expect no streaming
                if ok {
                    t.Fatalf("expected fallback, but got ok=true for %T", tc.in)
                }
 
```